roles/preflight-checks/tasks/main.yml

```diff
@@ -1,27 +1,24 @@
 ---
-- name: set os fact for rhel operating system
+- name: set ursula_os if undefined 
   set_fact:
-    os: rhel
-  when: ansible_distribution in ['CentOS', 'RedHat']
+    ursula_os: "{{ ansible_distribution | regex_replace('RedHat|CentOS', 'rhel')|lower }}"
+  when: ( ursula_os is undefined ) and
+        ( ansible_distribution in ['CentOS', 'RedHat'] or ansible_distribution == 'Ubuntu')
 
-- name: set os fact for rhel operating system
-  set_fact:
-    os: ubuntu
-  when: ansible_distribution in ['Ubuntu']
-
-- name: make sure os is defined
-  fail: msg="We don't support {{ ansible_distribution }} yet. OS should be CentOS, RedHat or Ubuntu."
-  when: os is undefined
+- name: validate ursula_os is a supported operating system and matches host system 
+  fail: msg="ursula_os does not match host operating system or {{ ursula_os }} is not a supported operating system (must be rhel or ubuntu)"
+  when: (ursula_os is defined) and ( ursula_os not in ['rhel', 'ubuntu'] or
```

Comment:
Since we are allowing the ursula_os var to be defined from envs, it's possible someone could set the fact to the wrong value. If someone were to define 'ubuntu' in an env and run ansible against a 'rhel' system this check would catch that.